### PR TITLE
Internal review of create vesting hook dapp

### PR DESCRIPTION
**Pull request by JeanNeiverth**
### How to test

- go to [dev.swap.cow.fi](https://dev.swap.cow.fi/#/100/swap/hooks)
- connect to a chain with deployed LlamaPay vesting factory (Mainnet, Gnosis or Arbitrum)
- go to: Add Post-Hook Action > My Custom Hooks > Add custom hook > type
`https://cow-hooks-dapps-create-vesting-git-jean-co-e5888a-bleu-builders.vercel.app` > add
- Add a new create vesting hook in the swap page

Note: as the bundle tenderly simulations feature is not merged into CoW develop, the simulations will run, but won't consider the swap state changes.

### CODE REVIEW

**Comment by yvesfracari** [comment]
Great work @JeanNeiverth Here is my review:

1- Hook details: https://snipboard.io/ayWZQs.jpg
2- The scrollbar package that you are using has a bug with Windows users. I suggest using the same as we use on the withdraw hook (we can even create a root layout default component with it)
3- Personal: for me it is more logical for the default option to be the first one, so I would put the `Input vesting amount manually` to the end of the list.
4- Instead of copping to the clipboard the Llamapay app link, I would just open it on a separate tab using a simple `<a target="_blank"></a>`.
5- Nitpick/personal, but I think that makes sense to me all the line of the radio inputs be clickable.
6- Disable the Submit button if fields are not filled.
7- Personal: Since all the other fields doesn't has a default option, I would remove default value of the period fields as well.
8- Nitpick: I would add a maximum value for the period numeric field
9- The submission can take a while to resolve ens names. I would use the `isSubimitting / isValidating` form state to show a loading text on the button for the user.
10- On declining approves it seems to auto-try again, I 
11- Nitpick: 
image
12- I changed this message a little bit on the withdraw dapp I think, could you take a look? I think that the address is on a tooltip. 
image
13- It failed without the error message and buttons when I tried to use the swap. I think that is because it is falling on estimating gas. I know that we're still trying to understand how to handle this case but I would copy the withdraw hook behavior for now (showing an error).
14- Nitpick: Change "Post-hook" to "post-hook" on button text.
15- They had requested to add `p-[16px]` on the withdraw hook, I would add it here as well.

**Review by mendesfabio — COMMENTED** [review]
great work @JeanNeiverth! some feedback I have:

- I'm having some trouble with the ENS - can you look into it? Just write an invalid ENS like `foo.bar` and try to add the hook, the input stops responding. 

- You can ask Bibi to provide better titles & description for each hook. At the moment I think they're too GPT'd, instead they should highlight the hooks' feature/functionality.

- IIUC this field indicates the time will take for all tokens to stream to the user? In that case, Vesting Duration/Period makes more sense. 

image


- I think the error message and buttons should be below the steps (only Approve hooks in this case). Also the correct is "on behalf" (instead of IN) but you could also simplify the phrase to: Approve proxy to execute hooks on your behalf.
image

- You could make this info looks more similar to CoW's
image
image

**Comment by ribeirojose** [comment]
Great stuff, @JeanNeiverth ! Thank you 🙇  Here's my comments:

## Copy
Screenshot 2024-10-21 at 4 34 40 PM
On the Vesting Amount input, should this be "Current amount"? 

## UI
Screenshot 2024-10-21 at 4 35 40 PM
The input under "Vesting amount" is not clickable/typeable, `#would` change its coloring a bit to denote this, perhaps even CoW already has something implemented.

Screenshot 2024-10-21 at 4 33 33 PM
Not sure how/why, but the bullet 'dot' is off center 😅 

Screenshot 2024-10-21 at 4 33 08 PM
Not sure if intended, but the component padding is larger on X than on Y.

Screenshot 2024-10-21 at 4 33 01 PM
Focus state is not using CoW's colors. Also it's not following the component borders. (If active/hover/focus are not clear, see [this](https://tailwindcss.com/docs/hover-focus-and-other-states#hover-focus-and-active).

Screenshot 2024-10-21 at 4 45 58 PM
Warning is looking a bit weird / malaligned / not in the same standard from other CoW warnings. Would get other references to see how they implement this but _perhaps_ we leave that warnings as a subtitle there instead of just showing them when a user selects that specific radio.

Screenshot 2024-10-21 at 4 33 16 PM
Inputs don't have the same left alignment spacing.

## UX
Screenshot 2024-10-21 at 4 33 56 PM
here you could highlight the link with an underscore and do an anchor tag instead of copying the URL. Perhaps have an arrow (upper right e.g. ↗️ ) so when the user clicks the browser will pop up the page in a new tab.

Screenshot 2024-10-21 at 4 44 52 PM
Can't click on text to select radio item. You can use [group](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state) to implement this.

## Bugs
Screenshot 2024-10-21 at 4 32 09 PM
After typing my own ENS, I can't interact with this input anymore.

File: apps/create-vesting/src/hooks/useGetHooksInfoVestUserAmount.ts [hunk 1]
```diff
@@ -23,7 +23,9 @@ export const useGetHooksInfoVestUserAmount = () => {
 
       if (!context?.account || !cowShedProxy || !amount) return;
 
-      const periodInSeconds = period * scaleToSecondsMapping[periodScale];
+      const periodInSeconds = Math.ceil(
+        period * scaleToSecondsMapping[periodScale],
+      );
```
Comment (yvesfracari) [inline apps/create-vesting/src/hooks/useGetHooksInfoVestUserAmount.ts:28]: I know that this logic is simple but it seems to repeat multiple times on the code. I would separate it in new file.

File: packages/cow-hooks-ui/tailwind.config.base.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ module.exports = {
           "rgb(var(--cow-color-success-text) / <alpha-value>)",
 
         "color-info": "rgb(var(--cow-color-info) / <alpha-value>)",
-        "color-info-bg": "rgb(var(--cow-color-info-bg) / <alpha-value>)",
+        "color-info-bg": "rgb(var(--cow-color-info-bg) / 0.15)",
```
Comment (yvesfracari) [inline packages/cow-hooks-ui/tailwind.config.base.js:81]: If you revert this change you can still use `bg-color-info-bg/15`

File: apps/create-vesting/src/components/RecipientInput.tsx [hunk 3]
```diff
@@ -1,13 +1,57 @@
 import { Input } from "@bleu/cow-hooks-ui";
+import { useState } from "react";
+
+export const RecipientInput = ({ value }: { value: string }) => {
+  const [displayValue, setDisplayValue] = useState(value);
+  const [isOnFocus, setIsOnFocus] = useState(false);
```
Comment (yvesfracari) [inline apps/create-vesting/src/components/RecipientInput.tsx:6]: Can you explain to me what is the behavior you are trying to implement here? I am wondering if there is a simpler way to do it

File: apps/create-vesting/src/app/page.tsx [hunk 4]
```diff
@@ -98,7 +100,7 @@ export default function Page() {
 
   if (!context?.orderParams?.buyTokenAddress)
     return (
-      <span className="mt-10 text-center">Provide a buy token in swap</span>
+      <span className="mt-10 text-center">Provide a buy amount in swap</span>
```
[the remark this answers is not in the document]
Comment (JeanNeiverth) [reply apps/create-vesting/src/app/page.tsx:103]: The issue is just that the buyTokenAddress in context is only updated f you provide both the token in swap and an amount. If you provide just the token, for some reason the context doesn't update it. I will change this to check the buyAmount directly

File: apps/create-vesting/src/app/page.tsx [hunk 5]
```diff
@@ -151,19 +161,26 @@ const InfoContent = () => {
...
+        target="_blank"
+        rel="noreferrer"
+        className="text-color-link underline"
+      >
+        llamapay.io/vesting
+        <ArrowTopRightIcon className="size-4 shrink-0 inline" />
+      </a>
     </span>
   );
 };
```
[the remark this answers is not in the document]
Comment (JeanNeiverth) [reply apps/create-vesting/src/app/page.tsx:175]: Actually, it is just the HTML I would insert into the return of the app. I made the Info.tsx component flexible to accept both strings or ReactNodes insides. In other words, this InfoContent is more like a constant string than a reusable component
